csrc/scheduler/mma_utils.h

```diff
@@ -271,10 +305,15 @@ MatmulProblemLayoutOpt getMmaLayout(Fusion* fusion);
 //!  be gathered.
 //!  TODO: 4th domain must be added for batch gemm support.
 ProblemIterDomainsOpt getProblemIterDomains(Fusion* fusion);
+ProblemIterDomainsOpt getProblemIterDomains(
+    mma_utils::MulSumProperties::InputsOutputs props);
 
 //! Returns wrapped collection of TensorView roles in fusion.
 //!  An error message is stored in retruned object if valid data cannot
 //!  be gathered.
+RolesMapOpt getTensorsRoles(
+    Fusion* fusion,
+    mma_utils::MulSumProperties::InputsOutputs props);
```

Comment:
Nitpick:

`mma_utils::MulSumProperties::InputsOutputs` is a simple structure but perhaps there is no need to make a copy of it when this helper function is called?